### ISSUE TITLE
fb_epid: I/O Intr not valid (no DSET)

### ISSUE THREAD

**Issue by prjemian** [epics-modules/optics#10]
from https://github.com/prjemian/epics-docker/issues/43

> after iocInit: `PV: xxx:epid1:outpv scanAdd: I/O Intr not valid (no DSET)`

The code in the [`fb_epid.db`](https://github.com/epics-modules/optics/blob/master/opticsApp/Db/fb_epid.db) database has this:

```
grecord(sseq, "$(P):outpv") {
        field(DESC, "follow OUT PV name")
        field(SCAN, "I/O Intr")
        field(DOL1, "$(P):out.OUTN")
        field(DLY1, "0")
        field(LNK1, "$(P):resume.INBN")
}
```

The `SCAN` field is triggering the message above.

**Comment by timmmooney** [comment]
This is the expected result of setting the SCAN field of a sseq record to "I/O Intr".  I/O Intr scanning requires device support that has implemented code described here: https://epics.anl.gov/base/R3-14/12-docs/AppDevGuide/node18.html#12357.  The sseq record has no device support that I know of.

Tim Mooney ***@***.***) (630)252-5417
Beamline Controls Group (www.aps.anl.gov)
Advanced Photon Source, Argonne National Lab

________________________________
From: Pete R Jemian ***@***.***>
Sent: Thursday, May 13, 2021 3:58 PM
To: epics-modules/optics ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [epics-modules/optics] fb_epid: I/O Intr not valid (no DSET) (#10)


from prjemian/epics-docker#43<https://github.com/prjemian/epics-docker/issues/43>

after iocInit: PV: xxx:epid1:outpv scanAdd: I/O Intr not valid (no DSET)

The code in the fb_epid.db<https://github.com/epics-modules/optics/blob/master/opticsApp/Db/fb_epid.db> database has this:

grecord(sseq, "$(P):outpv") {
        field(DESC, "follow OUT PV name")
        field(SCAN, "I/O Intr")
        field(DOL1, "$(P):out.OUTN")
        field(DLY1, "0")
        field(LNK1, "$(P):resume.INBN")
}


The SCAN field is triggering the message above.

**Comment by MarkRivers** [comment]
Note that the swait record in that database also have SCAN=I/O Intr.  Is that also an error?

How did this database ever work?

Should these lines all be deleted so that it defaults to SCAN=PASSIVE?

**Comment by keenanlang** [comment]
From the comments in the code, it appears that the intended function would be to set

$(P):outpv.SCAN = PASSIVE

and

$(P):out.FLNK = "$(P):outpv"

**Comment by prjemian** [comment]
I could try to test this in my [simulation IOC](https://github.com/prjemian/epics-docker) by modifying the database locally.

**Comment by prjemian** [comment]
Looks as if this `fb_epid` support is ripe for an overhaul.  We could compare with current use at 33ID.

**Comment by prjemian** [comment]
Also, there's no button for fb_epid.adl on the XXX green button screen.  `$(P)epid1`

**Comment by prjemian** [comment]
> Could be tested by constructing a _temperature controller_ simulation.

Indeed: such simulation is noted in the database: https://github.com/epics-modules/optics/blob/fdf5bc3c2731ba6769b62e71628fe3018c8245e3/opticsApp/Db/fb_epid.db#L236-L270

**Comment by prjemian** [comment]
Time for an overhaul of the `fb_epid` support.
